### PR TITLE
IGNITE-17871 JDKMarshaller replaced with OptimizedMarshaller for RAFT commands.

### PR DESCRIPTION
**Pull request by ibessonov**
https://issues.apache.org/jira/browse/IGNITE-17871

### CODE REVIEW

File: modules/raft/src/integrationTest/java/org/apache/ignite/raft/server/counter/GetValueCommand.java [hunk 1]
```diff
@@ -18,9 +18,14 @@
 package org.apache.ignite.raft.server.counter;
 
 import org.apache.ignite.internal.raft.ReadCommand;
+import org.apache.ignite.network.annotations.Transferable;
 
 /**
  * Get a value command.
  */
-public class GetValueCommand implements ReadCommand {
+@Transferable(1_000)
```
Comment (tkalkirill) [inline modules/raft/src/integrationTest/java/org/apache/ignite/raft/server/counter/GetValueCommand.java:26]: We do not have a single place where all types of messages?

File: modules/raft/src/main/java/org/apache/ignite/internal/raft/server/impl/JraftServerImpl.java [hunk 2]
```diff
@@ -113,6 +114,9 @@ public class JraftServerImpl implements RaftServer {
     /** Request executor. */
     private ExecutorService requestExecutor;
 
+    /** Marshaller. */
```
Comment (tkalkirill) [inline modules/raft/src/main/java/org/apache/ignite/internal/raft/server/impl/JraftServerImpl.java:117]: Marshaller for what?

File: modules/raft/src/integrationTest/java/org/apache/ignite/raft/server/counter/GetValueCommand.java (hunk 1 above)
Comment (ibessonov) [reply modules/raft/src/integrationTest/java/org/apache/ignite/raft/server/counter/GetValueCommand.java:26]: No, we don't, I'll do it if you're asking

File: modules/raft/src/main/java/org/apache/ignite/internal/raft/server/impl/JraftServerImpl.java (hunk 2 above)
Comment (ibessonov) [reply modules/raft/src/main/java/org/apache/ignite/internal/raft/server/impl/JraftServerImpl.java:117]: I'll fix it

**Review by tkalkirill — APPROVED** [review]

File: modules/raft/src/testFixtures/java/org/apache/ignite/raft/server/counter/IncrementAndGetCommand.java [hunk 3]
```diff
@@ -18,29 +18,22 @@
...
+@Transferable(TestMessageGroup.INCREMENT_AND_GET_COMMAND)
+public interface IncrementAndGetCommand extends WriteCommand {
     /**
      * Returns the delta.
      */
-    public long delta() {
-        return delta;
+    long delta();
+
+    static IncrementAndGetCommand incrementAndGetCommand(long delta) {
```
Comment (vldpyatkov) [inline modules/raft/src/testFixtures/java/org/apache/ignite/raft/server/counter/IncrementAndGetCommand.java:36]: Why do you use a static method to create an instance, if there is a factory?
This looks like contradictory patterns to me.

File: modules/raft/src/testFixtures/java/org/apache/ignite/raft/server/counter/IncrementAndGetCommand.java [hunk 4]
```diff
@@ -18,29 +18,22 @@
...
-    /**
-     * Constructor.
-     *
-     * @param delta The delta.
-     */
-    public IncrementAndGetCommand(long delta) {
-        this.delta = delta;
-    }
-
+//TODO Move to integration test directory when Maven build is not supported anymore.
```
Comment (vldpyatkov) [inline modules/raft/src/testFixtures/java/org/apache/ignite/raft/server/counter/IncrementAndGetCommand.java:28]: There is "todo" without a ticket number.

Comment (ibessonov) [reply modules/raft/src/testFixtures/java/org/apache/ignite/raft/server/counter/IncrementAndGetCommand.java:28]: I'll add it, thank you

File: modules/raft/src/testFixtures/java/org/apache/ignite/raft/server/counter/IncrementAndGetCommand.java (hunk 3 above)
Comment (ibessonov) [reply modules/raft/src/testFixtures/java/org/apache/ignite/raft/server/counter/IncrementAndGetCommand.java:36]: Just to be clear, these are test messages. I don't do things like that in production messages.

**Review by vldpyatkov — APPROVED** [review]

**Pull request merged**